dimod/serialization/bson.py

```diff
@@ -13,31 +13,43 @@
 #    limitations under the License.
 #
 # ================================================================================================
+from __future__ import absolute_import
 import itertools
 
 import numpy as np
 
 from dimod.binary_quadratic_model import BinaryQuadraticModel
 
+from six import PY2
+
 
 def bqm_bson_encoder(bqm):
     """todo"""
+
+    if PY2:
+        try:
+            import bson
+        except ImportError:
+            raise ImportError("Package `bson` from `pymongo` required for "
+                              "bson encoding in python 2")
+
     num_variables = len(bqm)
 
     index_dtype = np.uint32
     if num_variables <= 2**16:
         index_dtype = np.uint16
 
-    variable_order = sorted(bqm.linear)
-    num_possible_edges = num_variables*(num_variables - 1) // 2
-    density = len(bqm.quadratic) / num_possible_edges
+    variable_order = list(bqm.variables)
```

Comment:
If we want serialization to be unique (ie return the same for different "orderings" of bqm.variables), we should have a consistent ordering of the variables labels here.

Reply:
`sorted(bqm.variables, key=hash)`